# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/nsINode.cpp
# Commit: 48fc3f91f020
# Full Hash: 48fc3f91f0207ca38b1b6ab88f33c7c7c999d152
# Author: Serban Stanca <sstanca@mozilla.com>
# Date: 2025-05-09 04:15:00
# Regressor Bug: 1932150
# File Overlap Count: 1
# Description:
#   Revert "Bug 1932150 - Allow using flat tree order for point comparing in selection r=jjaschke,smaug,dom-core"
#   
#   This reverts commit 8d55fdef16bcd3e93e2aab7e4aead025c2d07ae4.
# ==============================================================================

diff -r 260574c2a78e -r 48fc3f91f020 dom/base/nsINode.cpp
--- a/dom/base/nsINode.cpp	Thu May 08 22:53:06 2025 +0300
+++ b/dom/base/nsINode.cpp	Thu May 08 22:53:07 2025 +0300
@@ -343,28 +343,16 @@
   }
 
   int operator()(const AbstractRange* const aRange) const {
-    auto ComparePoints = [](const nsINode* aNode1, const uint32_t aOffset1,
-                            const nsINode* aNode2, const uint32_t aOffset2,
-                            nsContentUtils::NodeIndexCache* aCache) {
-      if (StaticPrefs::dom_shadowdom_selection_across_boundary_enabled()) {
-        return nsContentUtils::ComparePointsWithIndices<TreeKind::Flat>(
-            aNode1, aOffset1, aNode2, aOffset2, aCache);
-      }
-      return nsContentUtils::ComparePointsWithIndices<
-          TreeKind::ShadowIncludingDOM>(aNode1, aOffset1, aNode2, aOffset2,
-                                        aCache);
-    };
-
-    Maybe<int32_t> cmp = ComparePoints(
-        &mNode, mEndOffset, aRange->GetMayCrossShadowBoundaryStartContainer(),
-        aRange->MayCrossShadowBoundaryStartOffset(), mCache);
-    MOZ_ASSERT(cmp.isSome());  // Should always be connected at this point.
-    if (cmp.value() == 1) {
-      cmp = ComparePoints(&mNode, mStartOffset,
-                          aRange->GetMayCrossShadowBoundaryEndContainer(),
-                          aRange->MayCrossShadowBoundaryEndOffset(), mCache);
-      MOZ_ASSERT(cmp.isSome());
-      if (cmp.value() == -1) {
+    Maybe<int32_t> cmp = nsContentUtils::ComparePoints(
+        ConstRawRangeBoundary(&mNode, mEndOffset,
+                              RangeBoundaryIsMutationObserved::No),
+        aRange->MayCrossShadowBoundaryStartRef(), mCache);
+    if (cmp.valueOr(1) == 1) {
+      cmp = nsContentUtils::ComparePoints(
+          ConstRawRangeBoundary(&mNode, mStartOffset,
+                                RangeBoundaryIsMutationObserved::No),
+          aRange->MayCrossShadowBoundaryEndRef(), mCache);
+      if (cmp.valueOr(1) == -1) {
         return 0;
       }
       return 1;